woudc_data_registry/processing.py

```diff
@@ -748,7 +748,8 @@ def check_instrument(self):
             self.extcsv.extcsv['INSTRUMENT']['Number'] = serial = 'UNKNOWN'
 
         instrument = build_instrument(self.extcsv)
-        fields = ['name', 'model', 'serial', 'station_id', 'dataset_id']
+        fields = ['name', 'model', 'serial',
+                  'station_id', 'dataset_id', 'deployment_id']
```

Comment:
The `fields` variable here has a specific meaning, it has to do with what SQL fields are filtered in the Registry to find a model instance without its identifier.

This part might be complicated to change, so look through all the methods in `registry.py` that are being called and understand what they all do before coming back to this.

Reply:
Adding on to this: having both `deployment_id` and `station_id` is redundant (why?) The best idea is probably to go name, model, serial, dataset, deployment in that list.